CDP4JsonSerializer/AutoGenDtoSerializer/PersonPermissionSerializer.cs

```
@@ -30,87 +29,429 @@
-        private readonly Dictionary<string, Func<object, JToken>> propertySerializerMap = new Dictionary<string, Func<object, JToken>>
+        /// <param name="thing">The <see cref="Thing" /> that have to be serialized</param>
+        /// <param name="writer">The <see cref="Utf8JsonWriter" /></param>
+        /// <param name="requestedDataModelVersion">The <see cref="Version" /> that has been requested for the serialization</param>
+        /// <exception cref="ArgumentException">If the provided <paramref name="thing" /> is not an <see cref="PersonPermission" /></exception>
+        /// <exception cref="NotSupportedException">If the provided <paramref name="requestedDataModelVersion" /> is not supported</exception>
+        public void Serialize(Thing thing, Utf8JsonWriter writer, Version requestedDataModelVersion)
         {
-            { "accessRight", accessRight => new JValue(accessRight.ToString()) },
-            { "actor", actor => new JValue(actor) },
-            { "classKind", classKind => new JValue(classKind.ToString()) },
-            { "excludedDomain", excludedDomain => new JArray(excludedDomain) },
-            { "excludedPerson", excludedPerson => new JArray(excludedPerson) },
-            { "iid", iid => new JValue(iid) },
-            { "isDeprecated", isDeprecated => new JValue(isDeprecated) },
-            { "modifiedOn", modifiedOn => new JValue(((DateTime)modifiedOn).ToString("yyyy-MM-ddTHH:mm:ss.fffZ")) },
-            { "objectClass", objectClass => new JValue(objectClass.ToString()) },
-            { "revisionNumber", revisionNumber => new JValue(revisionNumber) },
-            { "thingPreference", thingPreference => new JValue(thingPreference) },
-        };
+            if (thing is not PersonPermission personPermission)
+            {
+                throw new ArgumentException("The thing shall be a PersonPermission", nameof(thing));
+            }
+
+            if (requestedDataModelVersion < Version.Parse("1.0.0"))

```

can this ever happen, there is no version than 1.0.0. I think we should throw an exception since something is wrong

---

This was probably introduced for objects that have a minimum version > 1.0.0. Basically all CDP4 Extensions. Code should throw an error if the requestedDataModelVersion is < 1.0.0

---

resolved